examples/preprocessing/plot_find_ref_artifacts.py

```diff
@@ -0,0 +1,126 @@
+raw = io.read_raw_fif(raw_fname, preload=True)
+
+###############################################################################
+# Note that even though standard noise removal has already
+# been applied to these data, much of the noise in the reference channels
+# (bottom of the plot) can still be seen in the standard channels.
+raw.plot(duration=800, n_channels=160, scalings={"mag": 8e-13,
+                                                 "ref_meg": 2e-11})
+
+###############################################################################
+# The PSD of these data show the noise as clear peaks
+raw.plot_psd(fmax=30)
+
+###############################################################################
+# Run the "together" algorithm
+raw_tog = raw.copy()
+ica_kwargs = dict(
+    method='picard',
+    fit_params=dict(tol=1e-3),  # use a high tol here for speed
+)
+all_picks = mne.pick_types(raw_tog.info, meg=True, ref_meg=True)
+ica_tog = ICA(n_components=60, allow_ref_meg=True, **ica_kwargs)
+ica_tog.fit(raw_tog, picks=all_picks)
+bad_comps, scores = ica_tog.find_bads_ref(raw_tog, threshold=2.5)
+
+# Plot scores with bad components marked
+ica_tog.plot_scores(scores, bad_comps)
+
+# Examine the properties of removed components. It's clear from the time
+# courses and topographies that these components represent external,
+# intermittent noise.
+ica_tog.plot_properties(raw_tog, picks=bad_comps)
+
+# Remove the components
+raw_tog = ica_tog.apply(raw_tog, exclude=bad_comps)
+
+###############################################################################
+# Cleaned data
+raw_tog.plot_psd(fmax=30)
+
+###############################################################################
+# Now try the "separate" algorithm
+raw_sep = raw.copy()
+
+# Do ICA only on the reference channels
+ref_picks = mne.pick_types(raw_sep.info, meg=False, ref_meg=True)
+ica_ref = ICA(n_components=2, allow_ref_meg=True, **ica_kwargs)
+ica_ref.fit(raw_sep, picks=ref_picks)
+
+# Do ICA on both reference and standard channels. Here, we can just reuse
+# ica_tog from the section above
+ica_sep = ica_tog.copy()
+
+# Extract the time courses of these components and add them as channels
+# to the raw data. Think of them the same way as EOG/EKG channels, but instead
+# of giving info about eye movements/cardiac activity, they give info about
+# external magnetic noise
+ref_comps = ica_ref.get_sources(raw_sep)
+for c in ref_comps.ch_names:  # they need to have REF_ prefix to be recognised
+    ref_comps.rename_channels({c: "REF_" + c})
+raw_sep.add_channels([ref_comps])
+
+# Now that we have our noise channels, we run the separate algorithm.
+bad_comps, scores = ica_sep.find_bads_ref(raw_sep, method="separate")
+
+# Plot scores with bad components marked
+ica_sep.plot_scores(scores, bad_comps)
+
+# Examine the properties of removed components.
+ica_sep.plot_properties(raw_sep, picks=bad_comps)
+
+# Remove the components
+raw_sep = ica_sep.apply(raw_sep, exclude=bad_comps)
```

Comment:
Here again I'd be inclined to move this to the next code cell, to keep plotting code and plots adjacent.